File: AspNetMvcUnleashed/0672329980_source/Chapter07/VB/MvcApplication1/MvcApplication1/Controllers/ContentController.vb
Imports System.IO

Public Class ContentController
    Inherits System.Web.Mvc.Controller

    ' GET: /Content/Create
    Function Create() As ActionResult
        Return View()
    End Function

    ' POST: /Content/Create
    <AcceptVerbs(HttpVerbs.Post)> _
    Function Create(ByVal upload As HttpPostedFileBase) As ActionResult
        ' Save File
        Dim fileName = Path.GetFileName(upload.FileName)
        upload.SaveAs(Server.MapPath("~/Uploads/" & fileName))
        Return RedirectToAction("Create")
    End Function


End Class

File: AspNetMvcUnleashed/0672329980_source/Chapter07/CS/MvcApplication1/MvcApplication1/Controllers/ContentController.cs
using System.IO;
using System.Web;
using System.Web.Mvc;

namespace MvcApplication1.Controllers
{
    public class ContentController : Controller
    {
        // GET: /Content/Create
        public ActionResult Create()
        {
            return View();
        } 

        // POST: /Content/Create
        [AcceptVerbs(HttpVerbs.Post)]
        public ActionResult Create(HttpPostedFileBase upload)
        {
            // Save file
            var fileName = Path.GetFileName(upload.FileName);
            upload.SaveAs(Server.MapPath("~/Uploads/" + fileName));

            return RedirectToAction("Create");
        }
    }
}
